action.yml

```diff
@@ -14,6 +14,10 @@ inputs:
     description: 'Tailscale version to use.'
     required: true
     default: '1.32.1'
+  sha256sum:
```

Comment:
The SHA256 checksum can be obtained by appending ".sha256" to the download URL on pkgs.tailscale.com, for example:
https://pkgs.tailscale.com/stable/tailscale_1.38.4_arm64.tgz.sha256

The checksum could be checked automatically, not requiring a parameter. We would not add functionality which would break existing workflows by adding a new required parameter.

Reply:
What is the value of taking the sha256 as an input argument versus obtaining it via https://pkgs.tailscale.com/stable/tailscale_1.38.4_arm64.tgz.sha256 ? The only thing I can think of is protection from a compromised pkgs.tailscale.com, by specifying the sha256 which was known to be correct when entered.

One could make the argument optional, use it if provided, but fall back to checking https://pkgs.tailscale.com/stable/tailscale_1.38.4_arm64.tgz.sha256  if not provided. It would always provide the data integrity check, and could provide security reassurance if explicitly provided.

Reply:
> The only thing I can think of is protection from a compromised pkgs.tailscale.com, by specifying the sha256 which was known to be correct when entered.

That is the idea, yes.

> One could make the argument optional, use it if provided, but fall back to checking https://pkgs.tailscale.com/stable/tailscale_1.38.4_arm64.tgz.sha256 if not provided. It would always provide the data integrity check, and could provide security reassurance if explicitly provided.

That is a good idea. Thanks.

Reply:
https://github.com/tailscale/github-action/pull/63/commits/c7bfa08845653538483e2a0481d370377da06364